tests/trainer/test_fsdp.py

```diff
@@ -417,3 +403,54 @@ def oom_hook(module, grad_input, grad_ouput):
     output_2 = fsdp_oom_model(x)
 
     assert torch.equal(output_1, output_2)
+
+
+@pytest.mark.gpu
+@world_size(2)
+def test_fsdp_device_mesh(world_size: int):
+    model = SimpleModel()
+    model.fc1._fsdp_wrap = True  # pyright: ignore[reportGeneralTypeIssues]
+    model.fc2._fsdp_wrap = True  # pyright: ignore[reportGeneralTypeIssues]
+
+    # Expect warning via pytest
+    with pytest.warns(DeprecationWarning):
+        Trainer(
+            model=model,
+            parallelism_config={'fsdp': {
+                'device_mesh': [2],
+            }},
+            max_duration='3ba',
+        )
+
+
+@pytest.mark.gpu
+@world_size(2)
+def test_fsdp_shard(world_size: int):
```

Comment:
Are these tests intended just to test construction or should they actually test that the model is set up correctly?

Reply:
Just intended to test construction eg no errors with new API